src/actix/api/snapshot_api.rs

```diff
@@ -675,7 +675,31 @@ async fn recover_partial_snapshot_from(
     // nothing to verify
     let pass = new_unchecked_verification_pass();
 
+    let take_recovery_lock_future = async {
```

Comment:
same logic should be in regular `recover` method as well

Reply:
Maybe it would also me nice to make it a function and re-use

Reply:
Also, would it be possible to move this lock into Collection, so we won't even need to change API handlers?

Reply:
Not really (at least I don't see how), because we want to check this lock with `wait=false`, **before** we spawn recover_shard_snapshot future. It *can* be done, but will require refactoring recover snapshot methods.

Reply:
Done.